sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -370,8 +370,19 @@ ssize_t coap_build_reply(coap_pkt_t *pkt, unsigned code,
         return -ENOSPC;
     }
 
-    /* if code is COAP_CODE_EMPTY (zero), use RST as type, else RESP */
-    unsigned type = code ? COAP_RESP : COAP_RST;
+    /* if code is COAP_CODE_EMPTY (zero), assume Reset (RST) type */
+    unsigned type;
```

Comment:
minor suggestion: `unsigned type = COAP_TYPE_RST;` and omit the `else` below for readability. Otherwise looks good

Reply:
OK, no problem.